dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -1023,26 +1023,26 @@ private enum ToolType { TOOL, SERVICE, WORKFLOW, NOTEBOOK
         public ParsedRegistryID(String paramId) throws UnsupportedEncodingException {
             String id;
             id = URLDecoder.decode(paramId, StandardCharsets.UTF_8.displayName());
-            List<String> textSegments = Splitter.on('/').omitEmptyStrings().splitToList(id);
-            List<String> list = new ArrayList<>(textSegments);
-            String firstTextSegment = list.get(0);
+            List<String> segments = new ArrayList<>(Splitter.on('/').omitEmptyStrings().splitToList(id));
+            checkToolId(segments);
+            String firstTextSegment = segments.get(0);
             if (WORKFLOW_PREFIX.equalsIgnoreCase(firstTextSegment)) {
-                list.remove(0); // Remove #workflow from ArrayList to make parsing similar to tool
+                segments.remove(0); // Remove #workflow from ArrayList to make parsing similar to tool
                 type = ToolType.WORKFLOW;
             }
             if (SERVICE_PREFIX.equalsIgnoreCase(firstTextSegment)) {
-                list.remove(0); // Remove #service from ArrayList to make parsing similar to tool
+                segments.remove(0); // Remove #service from ArrayList to make parsing similar to tool
                 type = ToolType.SERVICE;
             }
             if (NOTEBOOK_PREFIX.equalsIgnoreCase(firstTextSegment)) {
-                list.remove(0); // Remove #notebook from ArrayList to make parsing similar to tool
+                segments.remove(0); // Remove #notebook from ArrayList to make parsing similar to tool
                 type = ToolType.NOTEBOOK;
             }
-            checkToolId(list);
-            registry = list.get(0);
-            organization = list.get(1);
-            name = list.get(2);
-            toolName = list.size() > SEGMENTS_IN_ID ? list.get(SEGMENTS_IN_ID) : "";
+            checkToolId(segments);
+            registry = segments.get(0);
+            organization = segments.get(1);
+            name = segments.get(2);
+            toolName = segments.size() > SEGMENTS_IN_ID ? segments.get(SEGMENTS_IN_ID) : "";
         }
```

Comment:
Check the tool id earlier, and refactor to improve clarity and remove a similar variable that could be accidentally used.